app/models/portal/teacher.rb

```diff
@@ -159,6 +162,15 @@ def my_classes_url(protocol, host)
     Rails.application.routes.url_helpers.api_v1_classes_mine(protocol: protocol, host: host)
   end
 
+  def add_recent_collection_page(project)
+    existing_rcp = self.teacher_project_views.where(viewed_projects_id: project.id).first
+    if existing_rcp.present?
+      existing_rcp.touch
+    else
+      self.viewed_projects << project
+    end
```

Comment:
I looked up some active record helpers for stuff like this and I think this could be simplified to:
```
project_view = self.teacher_project_views.where(viewed_project_id: project.id).first_or_create
project_view.touch
```
I switched this to using the singular `viewed_proect_id`. What it does is find the first project_view or create a new one if it doesn't exist. And then it touches this project_view.  The touch is a little inefficient when a project_view is first created, but that seems like it shouldn't be a problem.